src/main/java/com/thealgorithms/bitmanipulation/BitSwap.java

```diff
@@ -0,0 +1,19 @@
+package com.thealgorithms.bitmanipulation;
+
+public final class BitSwap {
+    private BitSwap() {
+    }
+    /*
+     * Swaps the bits at the position swap1 and swap2 from bit
+     */
+    public static int bitSwap(int bit, int swap1, int swap2) {
+        int val1 = (bit >> swap1) & 1;
+        int val2 = (bit >> swap2) & 1;
```

Comment:
What do you think about applying [`SingleBitOperations.getBit`](https://github.com/TheAlgorithms/Java/blob/1dc64b16859d95446a8bfe1c50692ce57316bc42/src/main/java/com/thealgorithms/bitmanipulation/SingleBitOperations.java#L31) here?

Reply:
I tought about it too, do you think it would be better to not use functions so it is easier to understand? If you think it is better we can do it.

Reply:
In my opinion it is easier to understand if `getBit` is used. Just have a look: `int bitAtPosA = (data >> posA) & 1` vs. `int bitAtPosA = SingleBitOperations.getBit(data, posA)`. The part `(data >> posA) & 1` is a bit _hacky_ and maybe needs a comment like _we extract the bit at position A_. But why not to simply use the method called `getBit`?

Moreover: `getBit` is tested. The line `(data >> posA) & 1` not.

Another advantage is that when you use `getBit` you can eliminate the variables and just have later the line `if (SingleBitOperations.getBit(data, posA) != SingleBitOperations.getBit(data, posB))`.

Reply:
I updated it.